PowershellTools.HostService/Program.cs

```diff
@@ -0,0 +1,119 @@
+                && args[2].StartsWith(Constants.ReadyEventUniqueNameArg, StringComparison.OrdinalIgnoreCase)
+                ))
+            {
+                return 1;
+            }
+
+            _processExitEvent = new AutoResetEvent(false);
+
+            string endpointGuid = args[0].Remove(0, Constants.UniqueEndpointArg.Length);
+            if (endpointGuid.Length != Guid.Empty.ToString().Length)
+            {
+                return 1;
+            }
+
+            int vsProcessId;
+            if (!Int32.TryParse(args[1].Remove(0, Constants.VsProcessIdArg.Length),
+                            NumberStyles.None,
+                            CultureInfo.InvariantCulture,
+                            out vsProcessId))
+            {
+                return 1;
+            }
+
+            string readyEventName = args[2].Remove(0, Constants.ReadyEventUniqueNameArg.Length);
+            if (readyEventName.Length < 36)
+            {
+                return 1;
+            }
+
+            // Step 1: Create the NetNamedPipeBinding. 
+            // Note: the setup of the binding should be same as the client side, otherwise, the connection won't get established
+            Uri baseAddress = new Uri(Constants.ProcessManagerHostUri + endpointGuid);
+            NetNamedPipeBinding binding = new NetNamedPipeBinding(NetNamedPipeSecurityMode.None);
+            binding.ReceiveTimeout = TimeSpan.MaxValue;
+            binding.Security.Transport.ProtectionLevel = ProtectionLevel.None;
+            binding.MaxReceivedMessageSize = Constants.BindingMaxReceivedMessageSize;
+
+            // Step 2: Create the service host.
+            CreatePowershellServiceHost(baseAddress, binding);
+            
+            // Step 3: Signal parent process that host is ready so that it can proceed.
+            EventWaitHandle readyEvent = new EventWaitHandle(false, EventResetMode.ManualReset, readyEventName);
+            readyEvent.Set();
+            readyEvent.Close();
+
+            try
+            {
+                // get parent process (The VS process)
+                Process p = Process.GetProcessById(vsProcessId);
+
+                if (p != null)
+                {
+                    p.EnableRaisingEvents = true;
+                    // Make sure the host process terminates when VS exits.
+                    p.Exited += new EventHandler(
+                        (sender, eventArgs) =>
+                        {
+                            if (_powershellServiceHost != null)
+                            {
+                                _powershellServiceHost.Close();
+                                _powershellServiceHost = null;
+                            }
+
+                            _processExitEvent.Set();
+                        });
+                }
+
+                _processExitEvent.WaitOne();
+            }
+            catch (Exception)
+            {
+

```

Comment:
Log/bubble up exceptions?  Even if it's a Debug.Writeline so they don't get mysteriously swallowed?